editor/components/post-text-editor/index.js

```diff
@@ -18,41 +18,45 @@ import { getEditedPostContent } from '../../store/selectors';
 import { editPost, resetBlocks, checkTemplateValidity } from '../../store/actions';
 
 class PostTextEditor extends Component {
-	constructor( props ) {
+	constructor() {
 		super( ...arguments );
 
-		this.onChange = this.onChange.bind( this );
-		this.onPersist = this.onPersist.bind( this );
+		this.handleFocus = this.handleFocus.bind( this );
+		this.handleChange = this.handleChange.bind( this );
+		this.handleBlur = this.handleBlur.bind( this );
 
 		this.state = {
-			initialValue: props.value,
+			value: null,
+			isDirty: false,
 		};
 	}
 
-	onChange( event ) {
-		this.props.onChange( event.target.value );
+	handleFocus() {
+		this.setState( { value: this.props.value } );
 	}
 
-	onPersist( event ) {
-		const { value } = event.target;
-		if ( value !== this.state.initialValue ) {
-			this.props.onPersist( value );
+	handleChange( event ) {
+		const value = event.target.value;
+		this.props.onChange( value );
+		this.setState( { value, isDirty: true } );
+	}
 
-			this.setState( {
-				initialValue: value,
-			} );
+	handleBlur() {
+		if ( this.state.isDirty ) {
```

Comment:
Do we need to track `isDirty`? Can't we just check `if ( this.state.value !== this.props.value ) {` ?

Reply:
Since we change the global value by dispatching `EDIT_POST` in our `onChange` handler, 
`this.state.value !== this.props.value` will always evaluate to false.